LinksAwakeningClient.py

```diff
@@ -563,17 +603,88 @@ async def server_auth(self, password_requested: bool = False):
+                        meta = self.check_name_to_metadata_map[name]
+                        check = self.client.tracker.get_check_from_meta(meta=meta)
+                        # If the check hasn't been collected yet, and should be
+                        if check is not None and check.id in self.client.tracker.remaining_checks \
+                                and name not in self.collected_blacklist:
+                            # Read to make sure the game has finished processing the last sent collection
+                            status = (await self.client.gameboy.async_read_memory_safe(
+                                LAClientConstants.wLinkCollectCheckValue))[0]
+                            # If not spin until either it has, or the game is won
+                            while (not (await self.client.is_victory())) and status != 0:
+                                time.sleep(0.1)
+                                status = (await self.client.gameboy.async_read_memory_safe(
+                                    LAClientConstants.wLinkCollectCheckValue))[0]
+                            # Split the check's memory address into high and low bytes, and send it to the game along
+                            # with the mask to apply
+                            high, low = divmod(check.address, 0x100)
+                            self.client.gameboy.write_memory(address=LAClientConstants.wLinkCollectCheckHigh,
+                                                             bytes=[high, low, check.mask])
+                            logger.info("Collecting check " + name)
+
+            self.examine_collected_checks = False
+
+    async def get_location_checks_from_server(self, checks: typing.List[int]):
+        return await self.send_msgs([{"cmd": "LocationScouts", "locations": checks}])
 
     async def sync(self):
         sync_msg = [{'cmd': 'Sync'}]
         await self.send_msgs(sync_msg)
 
     item_id_lookup = get_locations_to_id()
 
+    async def reset_game_loop(self):
+        """
+        Reset anything needed for the game loop to begin again.
+        """
+        # TODO: cancel all client tasks
+        if not self.client.stop_bizhawk_spam:
+            logger.info("(Re)Starting game loop")
+        self.found_checks.clear()
+        # On restart of game loop, clear all checks, just in case we swapped ROMs
+        # this isn't totally neccessary, but is extra safety against cross-ROM contamination
+        self.client.recvd_checks.clear()
+        await self.client.wait_for_retroarch_connection()
+        await self.client.reset_auth()
+        # If we find ourselves with new auth after the reset, reconnect
+        if self.auth and self.client.auth != self.auth:
+            # It would be neat to reconnect here, but connection needs this loop to be running
+            logger.info("Detected new ROM, disconnecting...")
+            await self.disconnect()
+            continue

```

Comment:
I'm testing this PR and this `continue` seems to be a problem when organized this way. 

I'm getting: `Syntax error 'continue' not properly in a loop`

It works for me when I put this block directly into `run_game_loop` as it was before.